modules/BTBidAdapter.js

```diff
@@ -0,0 +1,237 @@
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { isBoolean, isFn, isStr } from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'blockthrough';
+const GVLID = 815;
+const CURRENCY = config.getConfig('currency.adServerCurrency') || 'USD';
+const ENDPOINT_URL = 'https://pbs.btloader.com/openrtb2/auction';
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid: (bid) => {
+    return (
+      !!bid.params?.siteId &&
+      isStr(bid.params.siteId) &&
+      !!bid.params?.ab &&
+      isBoolean(bid.params.ab)
+    );
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    return validBidRequests.map((validBidRequest) => {
+      const request = getBTRequest(validBidRequest, bidderRequest);
+
+      return {
+        method: 'POST',
+        url: ENDPOINT_URL,
+        data: JSON.stringify(request),
+        bids: validBidRequests,
+      };
+    });
+  },
+
+  interpretResponse: (serverResponse, request) => {
+    if (!serverResponse) {
+      return [];
+    }
+
+    const seatBids = serverResponse?.body.seatbid || [];
+
+    if (!seatBids.length) {
+      return [];
+    }
+    const newBids = seatBids.flatMap((seat) => {
+      return seat.bid.map((bid) => {
+        const bidResponse = getNewBidResponse(bid);
+        bidResponse.btBidderCode = seat.seat;
+        return bidResponse;
+      });
+    });
+
+    return newBids;
+  },
+};
+
+function getNewBidResponse(bid) {
+  return {
+    requestId: bid.impid,
+    cpm: bid.price,
+    currency: bid.currency || CURRENCY,
+    width: bid.w || 350,
+    height: bid.h || 200,
+    ad: bid.adm,
+    ttl: bid.ttl || 360,
+    creativeId: bid.crid,
+    netRevenue: true,
+  };
+}
+
+function getBTRequest(validBidRequest, bidderRequest) {
+  let request = {
+    id: validBidRequest.auctionId,
+    source: getSource(validBidRequest),
+    imp: getImp(validBidRequest),
+    device: getDevice(validBidRequest),
+    site: getSite(validBidRequest),
+    user: getUser(validBidRequest, bidderRequest),
+    regs: getRegs(validBidRequest, bidderRequest),
+    at: 1,
+    ext: getExt(validBidRequest),
+  };
+
+  const isDebug = config.getConfig('debug') ? 1 : 0;
+  if (isDebug) {
+    request.test = isDebug;
+  }
+
+  return request;
+}
+
+function getImp(validBidRequest) {
+  const { params, mediaTypes, bidId, adUnitCode, ortb2Imp } = validBidRequest;
+  const { ab, siteId, ...bidderParams } = params;
+  const imps = [];
+
+  for (const mediaType in mediaTypes) {
```

Comment:
Personally I'm more of a fan of:
`const imps = Object.entries(mediaTypes).map(([mediaType, currentType]) => ...)`

Not really a huge deal either way; if you keep the `for ... in` I wonder if you'd need to add a `hasOwnProperty` check for safety

Reply:
do we expect `mediaType` to be anything other than a `banner`? 
I mean, you specified `supportedMediaTypes: [BANNER]`, but here if we get `VIDEO`, we won't error. am I missing something?

Reply:
Rn we only support banner, but in future we plan to support other types. Might be worth adding check for supported media types